src/main/java/com/hubspot/maven/plugins/dependency/management/DependencyManagementAnalyzer.java

```diff
@@ -24,22 +24,21 @@
   private boolean dependencyVersionDisallowedError = false;
   private List<String> errorMessages = new ArrayList<String>();
 
-  public DependencyManagementAnalyzer(MavenProject project, RequireManagement requireManagement, Log log) {
+  public DependencyManagementAnalyzer(MavenProject project, RequireManagement requireManagement, Consumer<String> logger) {
     this.project = project;
     this.requireManagement = requireManagement;
-    this.log = log;
+    this.logger = logger;
   }
 
   public boolean analyze() {
     boolean success = checkDependencyManagement();
     // don't combine with previous line, we don't want short-circuit evaluation
     success &= checkPluginManagement();
 
-
     if (!errorMessages.isEmpty()) {
-      log.warn("");
+      logViolation("");
       for(String msg : errorMessages) {
```

Comment:
Now that we're on java 8, this loop can be `errorMessage.forEach(logViolation);`